tests/redirects/base.py

```diff
@@ -103,7 +106,7 @@ def assert_valid_url(url, location=None, status_code=301, req_headers=None, req_
             # strip off query for further comparison
             resp_location = resp_location.split('?')[0]
 
-        assert resp_location == get_abs_url(location, base_url)
+        assert re.match(get_abs_url(location, base_url), resp_location) is not None
```

Comment:
I'm not sure we can trust this change w/o more changes to the tests. Every `.` needs escaping for example. Perhaps we could instead make using a regex an option, either by adding a boolean param (e.g. `regex=True`), or passing in a compiled regex as the test:

```python
url_test('/stuff/whatnot/', re.compile(r'/stfu/donnie'))
```